lisa/src/test/java/it/unive/lisa/test/imp/tutorial/intervals/Interval.java

```diff
@@ -0,0 +1,186 @@
+package it.unive.lisa.test.imp.tutorial.intervals;
+
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.BaseNonRelationalValueDomain;
+import it.unive.lisa.cfg.type.Type;
+import it.unive.lisa.symbolic.value.BinaryOperator;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.Identifier;
+import it.unive.lisa.symbolic.value.TernaryOperator;
+import it.unive.lisa.symbolic.value.UnaryOperator;
+
+public class Interval extends BaseNonRelationalValueDomain<Interval> {
```

Comment:
Why do we need a separate class `Intv` for the interval logic? It seems like a duplication.

If we merge those, the same considerations from the other classes above apply.